### PR TITLE
fix: ensure loading spinner stops when marketplace app installation fails

### LINKED ISSUE #35518: fix: Loading Stays Active When Clicking Outside Iframe Popup.
### **Description**
When clicking outside the "iframe" (anywhere else on the page), the loading indicator remains visible over the "Install" button. However, when closing the iframe using the close (X) button, this issue does not occur.

---

### **Steps to Reproduce**
1. Click on the three dots in the left pane.
2. Select **"Marketplace"**.
3. Choose an app that:
   - Has not been installed **OR**  
   - Has not been granted permission yet.  
4. A popup with an iframe appears.
5. Click **anywhere outside** the iframe/popup.  
6. Observe the loading indicator over the "Install" button.

---

### **Expected Behavior**
- The loading indicator should **not** remain visible over the "Install" button when clicking outside the iframe.
  
 #### Actual Behavior:

https://github.com/user-attachments/assets/c9004c61-6b87-4dba-8c08-964eeb0f1b8c

  
  #### Server/Client Information:
  
  #### **Server:**
  - **Operating System:** Windows 11 (using WSL Terminal)
  - **Rocket.Chat Server Version:** 7.5.0-develop
  - **Node.js Version:** 22.13.1
  - **Yarn Version:** 4.7.0
  
  #### **Client:**
  - **Client Type:** Browser (Microsoft Edge)
  - **Access URL:** [http://localhost:3000](http://localhost:3000)
  
  #### **Additional Information:**
  - **WSL Terminal:** Used for installing dependencies.
  
  ### **Are you working on this?**  
✅ Yes, I am working on this.

### LINKED ISSUE #38523: Install spinner stuck after failed marketplace install
### Description:

When installing an app from the Marketplace in a local development setup, the installation fails and the Install button spinner never stops, leaving the UI stuck in a loading state.

**Video** :

https://github.com/user-attachments/assets/46587706-0c45-41f0-b7bd-d6324d396d8b

### Steps to reproduce:

1. Run the server locally
2. Open Marketplace
3. Try installing any app
4. Installation fails
5. Install button spinner keeps spinning indefinitely

### Expected behavior:

The spinner should stop and show an error state if the app fails to install.

### Actual behavior:

The spinner continues spinning indefinitely even though the installation has already failed, and no error state is shown.

### Server Setup Information:

- Version of Rocket.Chat Server: 8.1.0-develop
- Operating System: Fedora Workstation 41
- Deployment Method: Local development
- NodeJS Version: 22.16.0

### Client Setup Information
-  Browser Version: Brave
- Operating System: Fedora Workstation 41

### Relevant logs:

TypeError: keysObject[onKeyDownEvent.key] is not a function
    at HTMLDocument.eval (ScreenshotCarouselAnchor.tsx:46:34)

### PR DESCRIPTION
**Pull request by dodaa08**
## Proposed changes (including videos or screenshots)

Fixes a UI issue where the Marketplace Install button spinner never stops when an app installation fails.
The loading state is now properly reset on failure so the spinner stops and the UI does not remain stuck.

https://github.com/user-attachments/assets/7b269cbc-2d70-438e-aa71-a0958309821a

## Issue(s)

Fixes #38523

Also Fixes #35518
A similar issue where spinner doesn't stop.
After:

https://github.com/user-attachments/assets/9bd13e25-d198-49f5-8324-851cb05d91cd

## Steps to test or reproduce

Run server locally

1. Open Marketplace
2. Try installing any app
3. Force installation failure
4. Spinner should stop and show error state

## Further comments

The fix uses a try-finally pattern which is the standard approach for ensuring cleanup code runs regardless of success/failure. We intentionally don't use try-catch because we want errors to propagate to the existing error handling infrastructure (React Query's 
onError
 → 
handleAPIError()
 → toast notification).

This is a minimal, focused fix that only addresses the loading state management without changing any business logic or error handling behavior.

[COMM-130]

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling and recovery mechanisms in marketplace app operations, ensuring more reliable and consistent loading state management throughout user interactions and application purchase transactions.

* **Tests**
  * Significantly expanded test coverage for marketplace app status operations with comprehensive test scenarios including multiple error handling paths, action workflows, edge cases, and state verification.

[COMM-130]: https://rocketchat.atlassian.net/browse/COMM-130?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by aleksandernsilva — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Could you also implement some unit tests covering this behavior?

**Comment by dodaa08** [comment]
> Thanks for the contribution! Could you also implement some unit tests covering this behavior?

Yeah sure, let me try

**Comment by dodaa08** [comment]
> Thanks for the contribution! Could you also implement some unit tests covering this behavior?

Hi, I've added unit tests to cover this behavior. I also made some changes to error handling in another file to gracefully handle errors and clear the loading state. Please review and let me know your thoughts.


Also for the tests I added it for 2-3 places so I used a global wrapper which can be used by all the places for mook app setup.

**Comment by dodaa08** [comment]
> Thanks for the contribution! Could you also implement some unit tests covering this behavior?

Hi, I've added them but this change feels a little big and I need some feedback on weather the tests seems fine, let me know once you review, thanks.

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 1]
```diff
@@ -1,31 +1,182 @@
...
 			}))
-			.wrap(mockedAppsContext)
-			.build(),
+			.wrap(mockedAppsContext);
+		return withAdminPermission ? builder.withPermission('manage-apps').build() : builder.build();
+	};
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+		(useMarketplaceActions as jest.Mock).mockReturnValue({
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]: I recommend using [jest.mocked](https://jestjs.io/docs/mock-function-api#jestmockedsource-options) utility function for these. You might need to adjust some types, but It will reduce the number of type casts necessary throughout the file.
Example:
```ts
const useMarketPlaceActionsMocked = jest.mocked(useMarketPlaceActions);
```
Usage: 
```ts
useMarketplaceActionsMocker.mockReturnValue()
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 2]
```diff
@@ -1,31 +1,182 @@
...
+		expect.assertions(3);
+		const app = createFakeApp({
+			installed: true,
+			version: '1.0.0',
+			marketplaceVersion: '2.0.0',
+		});
+		const updateError = new Error('update failed');
+		mockUpdate.mockRejectedValue(updateError);
+
+		let onSuccessCallback: (action: string, appPermissions?: unknown) => Promise<void>;
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]: `onSuccess` returns `void` not `Promise<void>`

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 3]
```diff
@@ -1,31 +1,182 @@
...
+
+		const button = screen.getByRole('button', { name: 'Update' });
+		await userEvent.click(button);
+		await act(async () => {
+			try {
+				await onSuccessCallback!('update');
+			} catch (err) {
+				expect(err).toBe(updateError);
+			}
+		});
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx:146]: The `await` here is forcing the error to be thrown within this try/catch but this changes the original component behavior, making the test less reliable. You should call `onSuccess` and let the component do the error handling. I don't think you'll need to assert the error in the final implementation. Validate the UI behavior not the internal component implementation.

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx [hunk 4]
```diff
@@ -56,15 +56,17 @@ const AppStatus = ({ app, showStatus = true, isAppDetailsPage, installed, ...pro
...
-					setEndUserRequested(true);
+			try {
+				if (action) {
+					if (action !== 'request') {
+						await marketplaceActions[action]({ ...app, permissionsGranted });
+					} else {
+						setEndUserRequested(true);
+					}
 				}
+			} finally {
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx:67]: I believe we are actually lacking error handling here. Although it works currently, the correct behavior would be to catch and handle the action error here and not propagate.  You can use the same helper `handleAPIError` here.

**Review by aleksandernsilva — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx (hunk 4 above)
Comment (dodaa08) [reply apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx:67]: **Sure**

image


Hoping these toasts won't show twice in production

https://github.com/user-attachments/assets/561c6a8f-bece-41d7-9d08-5a7c8c6ac49e


**Getting onto tests now.**

Comment (dodaa08) [reply apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx:67]: Hoping the checks should pass now, let me know if any code changes are needed.

**Review by dougfabris — CHANGES_REQUESTED** [review]
@dodaa08 we'll need a changeset here mate

**Comment by dodaa08** [comment]
> @dodaa08 we'll need a changeset here mate

Sure @dougfabris, let me push it

File: .changeset/gold-lies-cover.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@rocket.chat/meteor': patch
+---
+
+Fix marketplace app loading state display
```
Comment (aleksandernsilva) [inline .changeset/gold-lies-cover.md]:
```suggestion
Fixes marketplace's app install button getting stuck in a loading state when installation fails
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 6]
```diff
@@ -1,31 +1,181 @@
...
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+		mockedUseMarketplaceActions.mockReturnValue({
+			install: mockInstall,
+			update: mockUpdate,
+			purchase: mockPurchase,
+		});
+		mockedUseAppInstallationHandler.mockReturnValue(mockAppInstallationHandler);
+		mockedHandleAPIError.mockClear();
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 7]
```diff
@@ -1,31 +1,181 @@
...
-	render(<AppStatus app={app} showStatus isAppDetailsPage />, {
-		wrapper: mockAppRoot()
+describe('AppStatus', () => {
+	const mockInstall = jest.fn();
+	const mockUpdate = jest.fn();
+	const mockPurchase = jest.fn();
+	const mockAppInstallationHandler = jest.fn();
+	const mockedUseAppInstallationHandler = jest.mocked(useAppInstallationHandler);
+	const mockedUseMarketplaceActions = jest.mocked(useMarketplaceActions);
+	const mockedHandleAPIError = jest.mocked(handleAPIError);
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 8]
```diff
@@ -1,31 +1,181 @@
...
 import { mockAppRoot } from '@rocket.chat/mock-providers';
-import { render, screen } from '@testing-library/react';
+import { act, render, screen, waitFor } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 
 import AppStatus from './AppStatus';
 import { mockedAppsContext } from '../../../../../../tests/mocks/client/marketplace';
 import { createFakeApp, createFakeLicenseInfo } from '../../../../../../tests/mocks/data';
+import type { Actions } from '../../../helpers';
+import { handleAPIError } from '../../../helpers/handleAPIError';
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 9]
```diff
@@ -1,31 +1,181 @@
...
+				},
+				false,
+			),
+		});
+
+		expect(screen.getByRole('button', { name: 'Request' })).toBeInTheDocument();
+	});
+
+	it('should reset loading state when install action throws an error', async () => {
+		expect.assertions(3);
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]: Since you are using `waitFor` within the test, expecting a fixed number of assertions is unrealiable.
```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 10]
```diff
@@ -1,31 +1,181 @@
...
+			installed: false,
+			isPurchased: false,
+			price: 0,
+			purchaseType: 'buy',
+			versionIncompatible: false,
+		});
+		const installError = new Error('install failed');
+		mockInstall.mockRejectedValue(installError);
+
+		let onSuccessCallback: (action: Actions | '', appPermissions?: AppPermission[]) => void;
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
		let onSuccessCallback: (action: Actions | '', appPermissions?: AppPermission[]) => Promise<void>;
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 11]
```diff
@@ -1,31 +1,181 @@
...
+
+		render(<AppStatus app={app} showStatus isAppDetailsPage />, {
+			wrapper: createWrapper(),
+		});
+
+		const button = screen.getByRole('button', { name: 'Install' });
+		await userEvent.click(button);
+		await act(async () => {
+			onSuccessCallback('install');
+		});
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx:110]: Now that we can guarantee `onSuccess` returns a promise, we can go back to something similar to your initial commit. Also assert that the button has entered loading (disabled).
```suggestion
        
        expect(button).toBeDisabled();

		await act(async () => {
		    try {
				await onSuccessCallback('install');
		    } catch () {
		        expect(error).toBe(installError);
		    }
		});
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 12]
```diff
@@ -1,31 +1,181 @@
...
+		await waitFor(() => {
+			expect(button).not.toBeDisabled();
+		});
+		expect(mockInstall).toHaveBeenCalled();
+		expect(handleAPIError).toHaveBeenCalledWith(installError);
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]: catch already validates the error

```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 13]
```diff
@@ -1,31 +1,181 @@
...
+		const button = screen.getByRole('button', { name: 'Update' });
+		await userEvent.click(button);
+		await act(async () => {
+			onSuccessCallback('update');
+		});
+		await waitFor(() => {
+			expect(button).not.toBeDisabled();
+		});
+		expect(mockUpdate).toHaveBeenCalled();
+		expect(handleAPIError).toHaveBeenCalledWith(updateError);
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 14]
```diff
@@ -1,31 +1,181 @@
...
+		const button = screen.getByRole('button', { name: 'Buy' });
+		await userEvent.click(button);
+		await act(async () => {
+			onSuccessCallback('purchase');
+		});
+		await waitFor(() => {
+			expect(button).not.toBeDisabled();
+		});
+		expect(mockPurchase).toHaveBeenCalled();
+		expect(handleAPIError).toHaveBeenCalledWith(purchaseError);
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 15]
```diff
@@ -1,31 +1,181 @@
...
+		const app = createFakeApp({
+			installed: false,
+			isPurchased: false,
+			price: 10,
+			purchaseType: 'buy',
+		});
+		const purchaseError = new Error('purchase failed');
+		mockPurchase.mockRejectedValue(purchaseError);
+
+		let onSuccessCallback: (action: Actions | '', appPermissions?: AppPermission[]) => void;
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
		let onSuccessCallback: (action: Actions | '', appPermissions?: AppPermission[]) => Promise<void>;
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 16]
```diff
@@ -1,31 +1,181 @@
...
+
+		render(<AppStatus app={app} showStatus isAppDetailsPage installed />, {
+			wrapper: createWrapper({ installedApps: 1, totalMarketplaceEnabled: 1 }),
+		});
+
+		const button = screen.getByRole('button', { name: 'Update' });
+		await userEvent.click(button);
+		await act(async () => {
+			onSuccessCallback('update');
+		});
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx:146]:
```suggestion
        
        expect(button).toBeDisabled();

		await act(async () => {
		    try {
				await onSuccessCallback('update');
		    } catch () {
		        expect(error).toBe(updateError);
		    }
		});
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 17]
```diff
@@ -1,31 +1,181 @@
...
+		await waitFor(() => {
+			expect(button).not.toBeDisabled();
+		});
+		expect(mockInstall).toHaveBeenCalled();
+		expect(handleAPIError).toHaveBeenCalledWith(installError);
+	});
+
+	it('should reset loading state when update action throws an error', async () => {
+		expect.assertions(3);
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 18]
```diff
@@ -1,31 +1,181 @@
...
+		});
+		await waitFor(() => {
+			expect(button).not.toBeDisabled();
+		});
+		expect(mockUpdate).toHaveBeenCalled();
+		expect(handleAPIError).toHaveBeenCalledWith(updateError);
+	});
+
+	it('should reset loading state when purchase action throws an error', async () => {
+		expect.assertions(3);
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 19]
```diff
@@ -1,31 +1,181 @@
...
+
+		render(<AppStatus app={app} showStatus isAppDetailsPage />, {
+			wrapper: createWrapper(),
+		});
+
+		const button = screen.getByRole('button', { name: 'Buy' });
+		await userEvent.click(button);
+		await act(async () => {
+			onSuccessCallback('purchase');
+		});
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx:183]:
```suggestion
        expect(button).toBeDisabled();

		await act(async () => {
		    try {
				await onSuccessCallback('purchase');
		    } catch () {
		        expect(error).toBe(purchaseError);
		    }
		});
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx [hunk 20]
```diff
@@ -1,31 +1,181 @@
...
+		expect.assertions(3);
+		const app = createFakeApp({
+			installed: true,
+			version: '1.0.0',
+			marketplaceVersion: '2.0.0',
+		});
+		const updateError = new Error('update failed');
+		mockUpdate.mockRejectedValue(updateError);
+
+		let onSuccessCallback: (action: Actions | '', appPermissions?: AppPermission[]) => void;
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.spec.tsx]:
```suggestion
		let onSuccessCallback: (action: Actions | '', appPermissions?: AppPermission[]) => Promise<void>;
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx [hunk 21]
```diff
@@ -56,15 +57,19 @@ const AppStatus = ({ app, showStatus = true, isAppDetailsPage, installed, ...pro
...
+			try {
+				if (action) {
+					if (action !== 'request') {
+						await marketplaceActions[action]({ ...app, permissionsGranted });
+					} else {
+						setEndUserRequested(true);
+					}
 				}
+			} catch (error) {
+				handleAPIError(error);
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx]:
```suggestion
```

**Review by aleksandernsilva — CHANGES_REQUESTED** [review]
Hey, @dodaa08 . Thanks for your patience, sometimes implementing the tests takes a lot of effort, but it's worth it in the long run.

I noticed tests are suffering because the `onSuccess` typing and implementation is inconsistent. So let's update [AppInstallationHandlerParams](https://github.com/RocketChat/Rocket.Chat/blob/8f5df556dcc46001d6d6f60ce92f58e99b187dbd/apps/meteor/client/views/marketplace/hooks/useAppInstallationHandler.tsx#L21) so `onSuccess` matches the implementation and returns `Promise<void>`.

After that, you can use the suggestions I highlighted as a guide. In the end things should look similar to your initial implementation (try/catch) with a few additions.

After these adjustments we should be good to go 🚀 

PS.: Do not commit the suggestions blindly. Use them as a guide and test locally.

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx (hunk 4 above)
Comment (aleksandernsilva) [reply apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppStatus/AppStatus.tsx:67]: You are correct about the duplicated toasts. I also did a few tests here and for now I believe it's best to keep as it was (without the catch) as to not deviate to much from the scope of these PR. Let's focus our efforts and adjusting the tests.

**Comment by dodaa08** [comment]
Thanks @aleksandernsilva for all the suggestions, Ran locally Screenshot From 2026-02-12 01-06-37 Commited!, Let me know your thoughts, once you review.

**Comment by dodaa08** [comment]
@aleksandernsilva  I tried a few times but the tests are failing 

Screenshot From 2026-02-13 00-05-30

I think they were fine yesterday, but after a few merge they doesn't seem to pass.

**Comment by dodaa08** [comment]
Hi @aleksandernsilva , the checks are passing now, let me know if anything else is needed here, once you review.

**Comment by dodaa08** [comment]
> Hey, @dodaa08 . Thanks for your patience, sometimes implementing the tests takes a lot of effort, but it's worth it in the long run.
> 
> I noticed tests are suffering because the `onSuccess` typing and implementation is inconsistent. So let's update [AppInstallationHandlerParams](https://github.com/RocketChat/Rocket.Chat/blob/8f5df556dcc46001d6d6f60ce92f58e99b187dbd/apps/meteor/client/views/marketplace/hooks/useAppInstallationHandler.tsx#L21) so `onSuccess` matches the implementation and returns `Promise<void>`.
> 
> After that, you can use the suggestions I highlighted as a guide. In the end things should look similar to your initial implementation (try/catch) with a few additions.
> 
> After these adjustments we should be good to go 🚀
> 
> PS.: Do not commit the suggestions blindly. Use them as a guide and test locally.

Hi @aleksandernsilva , I waiting for your response.
